ServiceProviderBot/Bot/Dialogs/DialogBase.cs

```diff
@@ -12,14 +14,14 @@ public abstract class DialogBase
     {
         protected readonly StateAccessors state;
         protected readonly DialogSet dialogs;
-        protected readonly DbInterface database;
+        protected readonly IApiInterface api;
```

Comment:
Code now uses an abstracted interface for API calls. The bot passes in an interface for Common Data Service (CDS) and the tests pass in an interface for Entity Framework